### ISSUE TITLE
method hyperparam_search in class GaussianProcessHyperparamOpt throws error with Python 3

### ISSUE THREAD

**Issue by amandajean119** [deepchem/deepchem#1231]
In the method hyperparam_search in the class GaussianProcessHyperparamOpt, the following lines generate an **AttributeError: 'dict_keys' object has no attribute 'remove'**

```
hyper_parameters = params_dict
hp_list = hyper_parameters.keys()
for hp in hp_invalid_list:
  if hp in hp_list:
    hp_list.remove(hp)
```
In Python 3, dict.keys() produces a view object, rather than a list, as Python 2 does: https://docs.python.org/3/library/stdtypes.html#dict-views
This view object has no method remove. An easy fix is just casting this as a list:
`hp_list = list(hyper_parameters.keys())`

**Comment by miaecle** [comment]
@amandajean119 Good catch! This method is developed under python2 environment and I missed the compatibility issue here. Please feel free to initiate a PR to fix this!

**Comment by rbharath** [comment]
@amandajean119 Thanks for the fix! Closing since #1235 has been merged in.
